Detectors/ZDC/workflow/src/DigitRecoSpec.cxx

```diff
@@ -165,7 +174,20 @@ void DigitRecoSpec::run(ProcessingContext& pc)
   auto chans = pc.inputs().get<gsl::span<o2::zdc::ChannelData>>("chan");
   auto peds = pc.inputs().get<gsl::span<o2::zdc::OrbitData>>("peds");
 
-  mWorker.process(peds, bcdata, chans);
+  // Reduce load by dropping random time frames
+  bool toProcess = true;
+  if (mRecoFraction <= 0.) {
+    toProcess = false;
+  } else if (mRecoFraction < 1.0) {
+    double frac = std::rand() / double(RAND_MAX);
+    if (frac > mRecoFraction) {
+      toProcess = false;
+    }
+  }
+
+  if (toProcess) {
+    mWorker.process(peds, bcdata, chans);
+  }
   const std::vector<o2::zdc::RecEventAux>& recAux = mWorker.getReco();
```

Comment:
The DigitReco::mReco returned by `mWorker.getReco()` is reset in the `mWorker.process` call, so for the TFs which you don't want to process you will get the mReco of the previous TF. 
One should clear mReco in every TF.

Reply:
Corrected the compilation error. I overlooked it. Apologies.
Concerning this there was a protection in the loop by inserting a break if the event is stale. However the logic was indeed cumbersome and the printed info message was not clear. I changed to protect everything inside an if(toProcess). Now if the TF is skipped there is an explicit message and mReco is not used.